salt/modules/status.py

```diff
@@ -1369,7 +1370,9 @@ def sunos_netdev():
         """
         ret = {}
         ##NOTE: we cannot use hwaddr_interfaces here, so we grab both ip4 and ip6
-        for dev in __grains__["ip4_interfaces"].keys() + __grains__["ip6_interfaces"]:
+        for dev in itertools.chain(
+            __grains__["ip4_interfaces"].keys(), __grains__["ip6_interfaces"].keys()
```

Comment:
**suggestion(non-blocking)** FWIW, keys are unnecessary here. I was curious so I double checked:

```
import timeit
import itertools


a = {'foo': 'foo', 'bar': 'bar', 'bang': 'bang'}
b = {"1": 1, "2": 2}

def one():
    for i in itertools.chain(a,b):
        x = i[::-1]

def two():
    for i in itertools.chain(a.keys(), b.keys()):
        x = i[::-1]


print(timeit.timeit(one))
print(timeit.timeit(two))

# Results:
# 0.9081845579999026
# 0.9607179529998575
```

Even swapping the order of timeit it looks like the non-keys() version is slightly faster. Not enough that _I_ think it's a blocker, but if you want to make the change then :+1:

Reply:
Leave it as is, since working now.
But going to go through all of the code since found other Python 2'isms which are causing issue on the AIX side, and our tools are not catching this use of Python 2 which is illegal syntax in Python 3.
Will probably update the code in that PR